[Bug]: Improve clean-room ability to run the Data TCK

### Specification Version

1.0.0-RC1

### Bug report

The current state of the Data TCK could use some improvement regarding the ability of a party creating a clean-room implementation of the spec to run the TCK against their impl. The current state of the data TCK requires a fair amount of a priori knowledge to even run the TCK.

- No info about how persistence provider is supposed to be run.
- No info about having CDI ready.
- Missing info or persistence.xml descriptor.
- Unclear if annotation provided needs to be run.


### Additional information

_No response_

Here's my experience with "unboxing" of the TCK itself:

* I took the latest binary (the one with sha 02dda1...) and unzipped it
* found the readme in the root and opened the guide found in the `docs` folder
* going through the UG:
  * sect 3.1 says there is SE mode - great
  * sect 4.1 says EE Platform is required if running in EE Mode - I'm not interested in this, so I don't need the EE server
  * sect 5.2
    * `artifacts` contains The TCK test classes and _source_ - if this is supposed to mean that there is a _-tck-sources.jar_ in the TCK distribution then such jar is missing and either the guide or the zip is wrong
    * just noticed a typo - `doc` folder vs actual `docs` folder
    * a script.. let's run it:
```
➜  artifacts ./artifact-install.sh 
zsh: permission denied: ./artifact-install.sh
➜  artifacts chmod u+x artifact-install.sh 
➜  artifacts ./artifact-install.sh        
A valid version is required to execute this script
```
what? Why cannot the script _by default_ install files it has next to it? Never mind, let's pass `1.0.0` to the script and move on
  * sect 5.3.2 - SE Mode - this is what I'm looking for! let's jump to sect 7 and more specifically 7.1
  * sect 7.1 - the runner described here seems to be available already, so let's try it:
```
➜  starter mvn clean test -f se-pom.xml
...
[ERROR] 'groupId' with value '[YOUR_GROUP_ID]' does not match a valid id pattern. @ line 23, column 11
...
```
...interactive experience :-) let's set `gId` to `com.acme` and try again
```
➜  starter mvn clean test -f se-pom.xml
...
[ERROR] Tests run: 85, Failures: 73, Errors: 11, Skipped: 0
```
cool! There is at least one test which passes and that is the signature test. But wait a sec - the `se-pom` did run tests from some `jakarta.data:jakarta.data-tck:1.0.0-SNAPSHOT` and not from the tck jar installed by the `artifact-install.sh` (apparently not everything from #729 was addressed) and the version of the API was not `1.0.0` but `1.0.0-SNAPSHOT`

IDK but I expect that if I have some installer and some example then the example is bound to the artifact versions installed by provided installer and the only `-SNAPSHOT` I can find is in the runner's version.

...so let's fix the version of dependencies in the pom and try again - same results, but why is there new `[path-to-extract-jvm-classes]` folder under the starter? It belongs under the `target` folder, doesn't it? There are also few log files, well, in this case some people prefer to have them in the top level folder, so why not..

Now, when I look at the test failures themselves - all of them are caused by NPE, looking into sources reveals that NPE happens because injection does not happen (ie [here](https://github.com/jakartaee/data/blob/main/tck/src/main/java/ee/jakarta/tck/data/standalone/entity/EntityTests.java#L108)) - why the injection is even expected to work in SE mode? What does SE and/or standalone mode within the context of the Data TCK mean? The Data Spec seems to be silent about requiring CDI in non EE products (...but I can be wrong of course)

few other comments:
* consider updating maven plugin versions in poms of sample runners (surefire 2.22.2 is a bit outdated today)
* consider updating API dependencies to EE 11 versions (ie CDI 4.1.0, 6.1.0,...)
* consider satisfying the _MUST contain Instructions describing how to run the compatible implementation(s) that are being used to validate the TCK_ requirement of the TCK Process by inclusion instead of by reference to the [OpenLiberty fat tck](https://github.com/OpenLiberty/open-liberty/tree/integration/dev/io.openliberty.jakarta.data.1.0_fat_tck) (ie use jnosql or anything showing that it is possible to pass all defined tests)

--- a/.github/dependabot.yml
+++ b/.github/dependabot.yml
@@ -9,3 +9,9 @@ updates:
     directory: /
     schedule:
       interval: daily
+
+  # TODO not sure if dependabot will recognize non-standard pom file names? 
+  - package-ecosystem: maven
+    directory: /tck-dist/src/main/starter/
+    schedule:
+      interval: daily

--- a/.github/dependabot.yml
+++ b/.github/dependabot.yml
@@ -10,8 +10,7 @@ updates:
     schedule:
       interval: daily
 
-  # TODO not sure if dependabot will recognize non-standard pom file names? 
   - package-ecosystem: maven
-    directory: /tck-dist/src/main/starter/
+    directory: /tck-dist/src/main/
     schedule:
       interval: daily

--- a/tck-dist/pom.xml
+++ b/tck-dist/pom.xml
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!--
- ~ Copyright (c) 2022, 2023 Contributors to the Eclipse Foundation
+ ~ Copyright (c) 2022, 2024 Contributors to the Eclipse Foundation
  ~
  ~ This program and the accompanying materials are made available under the
  ~ terms of the Eclipse Public License v. 2.0, which is available at
@@ -87,17 +87,31 @@
     </configuration>
    </plugin>
 
-   <!-- Load TCK dependency location as variable -->
    <plugin>
     <groupId>org.apache.maven.plugins</groupId>
     <artifactId>maven-dependency-plugin</artifactId>
     <version>3.6.1</version>
     <executions>
+	<!-- Load TCK dependency location as variable -->
      <execution>
       <phase>initialize</phase>
       <goals>
        <goal>properties</goal>
       </goals>
+     </execution>
+     <!-- Copy tck sources to a known location -->
+     <execution>
+       <id>get-dependency-sources</id>
+       <phase>package</phase>
+       <goals>
+         <goal>copy-dependencies</goal>
+       </goals>
+       <configuration>
+         <includeArtifactIds>jakarta.data-tck</includeArtifactIds>
+         <classifier>sources</classifier>
+         <prependGroupId>true</prependGroupId>
+         <outputDirectory>${project.build.directory}/copied-sources/</outputDirectory>
+       </configuration>
      </execution>
     </executions>
    </plugin>
@@ -119,7 +133,7 @@
     <configuration>
      <mainClass>ee.jakarta.tck.data.metadata.CollectMetaData</mainClass>
      <arguments>
-      <argument>true</argument>
+      <argument>false</argument> <!-- debug -->
       <argument>${jakarta.data:jakarta.data-tck:jar}</argument>
       <argument>${project.basedir}/src/main/asciidoc/sections/generated</argument>
      </arguments>

--- a/tck-dist/src/main/artifacts/artifact-install.sh
+++ b/tck-dist/src/main/artifacts/artifact-install.sh
@@ -3,7 +3,7 @@
 ## A sample script to install the artifact directory contents into a local maven repository
 
 POM=$(ls *.pom)
-TCK=$(ls *.jar) ## FIXME might pick up sources.jar if not careful
+TCK=$(ls *.jar | grep -v sources)
 
 NO_EXT=${POM%.*}      # jakarta.data-parent-1.0.0-SNAPSHOT.pom > jakarta.data-parent-1.0.0-SNAPSHOT
 NO_REPO=${NO_EXT#*-}  # jakarta.data-parent-1.0.0-SNAPSHOT > parent-1.0.0-SNAPSHOT

--- a/tck-dist/src/main/artifacts/artifact-install.sh
+++ b/tck-dist/src/main/artifacts/artifact-install.sh
@@ -9,7 +9,7 @@ NO_EXT=${POM%.*}      # jakarta.data-parent-1.0.0-SNAPSHOT.pom > jakarta.data-pa
 NO_REPO=${NO_EXT#*-}  # jakarta.data-parent-1.0.0-SNAPSHOT > parent-1.0.0-SNAPSHOT
 VERSION=${NO_REPO#*-} # parent-1.0.0-SNAPSHOT > 1.0.0-SNAPSHOT
 
-echo "Installing $POM with version $VERSION
+echo "Installing $POM with version $VERSION"
 mvn org.apache.maven.plugins:maven-install-plugin:3.0.0-M1:install-file \
 -Dfile=$POM \
 -DgroupId=jakarta.data \
@@ -17,6 +17,6 @@ mvn org.apache.maven.plugins:maven-install-plugin:3.0.0-M1:install-file \
 -Dversion="$VERSION" \
 -Dpackaging=pom
 
-echo "Installing $TCK
+echo "Installing $TCK"
 mvn org.apache.maven.plugins:maven-install-plugin:3.0.0-M1:install-file \
 -Dfile=$TCK

--- a/tck-dist/src/main/artifacts/artifact-install.sh
+++ b/tck-dist/src/main/artifacts/artifact-install.sh
@@ -2,21 +2,21 @@
 
 ## A sample script to install the artifact directory contents into a local maven repository
 
-if [[ $1 =~ ^[0-9]+\.[0-9]+\.[0-9]+.*$ ]]; then
-  VERSION="$1"
-else
-  echo "A valid version is required to execute this script"
-  exit 1
-fi
+POM=$(ls *.pom)
+TCK=$(ls *.jar) ## FIXME might pick up sources.jar if not careful
 
-# Parent pom
+NO_EXT=${POM%.*}      # jakarta.data-parent-1.0.0-SNAPSHOT.pom > jakarta.data-parent-1.0.0-SNAPSHOT
+NO_REPO=${NO_EXT#*-}  # jakarta.data-parent-1.0.0-SNAPSHOT > parent-1.0.0-SNAPSHOT
+VERSION=${NO_REPO#*-} # parent-1.0.0-SNAPSHOT > 1.0.0-SNAPSHOT
+
+echo "Installing $POM with version $VERSION
 mvn org.apache.maven.plugins:maven-install-plugin:3.0.0-M1:install-file \
--Dfile=jakarta.data-parent-"$VERSION".pom \
+-Dfile=$POM \
 -DgroupId=jakarta.data \
 -DartifactId=jakarta.data-parent \
 -Dversion="$VERSION" \
 -Dpackaging=pom
 
-# Jakarta Data TCK Installed Library
+echo "Installing $TCK
 mvn org.apache.maven.plugins:maven-install-plugin:3.0.0-M1:install-file \
--Dfile=jakarta.data-tck-"$VERSION".jar
+-Dfile=$TCK

--- a/tck-dist/src/main/asciidoc/sections/05inventory.adoc
+++ b/tck-dist/src/main/asciidoc/sections/05inventory.adoc
@@ -67,6 +67,7 @@ Here is an essential checklist of what you will need per mode, and links to the 
 SE Mode:
 
 - The {APIShortName} API, JUnit5, and Signature Test Plugin libraries available at runtime | <<Standalone Dependencies>>
+- A dependency injection framework supported by your implementation | <<Standalone Dependencies>>
 - A logging configuration for TCK logging available at runtime | <<Standalone Logging>>
 
 EE Mode:

--- a/tck-dist/src/main/asciidoc/sections/05inventory.adoc
+++ b/tck-dist/src/main/asciidoc/sections/05inventory.adoc
@@ -16,7 +16,7 @@ Once the TCK is extracted, you'll see the following structure:
 ----
 data-tck-<version>-dist/
   artifacts/
-  doc/
+  docs/
   starter/
   LICENSE
   README.md
@@ -29,7 +29,7 @@ In more detail:
 ** The {TCKTestPlatform} configuration file
 ** A copy of the SignatureTest file for reference
 ** A script to copy the TCK into local maven repository.
-* `doc` contains the documentation for the TCK (i.e. this reference guide)
+* `docs` contains the documentation for the TCK (i.e. this reference guide)
 * `starter` a very basic starter maven project to get you started.
 
 === A Quick Tour of the TCK Artifacts

--- a/tck-dist/src/main/asciidoc/sections/08example-runner.adoc
+++ b/tck-dist/src/main/asciidoc/sections/08example-runner.adoc
@@ -7,8 +7,14 @@ Below are links to projects where the {APILongName} TCK is being used and run su
 |===
 |Project |Link |Profile(s)
 
+|Hibernate
+|https://github.com/jakartaredhat/hibernate-data-tck/tree/3e5fd4ddc68b60cc026987021d8d3e80ebed6809[hibernate-data-tck]
+|standalone
+
 |Open Liberty
-|https://github.com/OpenLiberty/open-liberty/tree/integration/dev/io.openliberty.jakarta.data.1.0_fat_tck
-|standalone, core, and web
+|https://github.com/OpenLiberty/open-liberty/tree/549a6b89005de62289df7b550aedd229dd407722/dev/io.openliberty.jakarta.data.1.0_fat_tck[open-liberty/io.openliberty.jakarta.data.1.0_fat_tck]
+|core, web, full
 
 |===
+
+// TODO add JNoSQL once implemented

--- a/tck-dist/src/main/asciidoc/sections/10signature.adoc
+++ b/tck-dist/src/main/asciidoc/sections/10signature.adoc
@@ -18,7 +18,7 @@ The {APILongName} TCK will run signature tests within a standalone or deployed a
 and not as part of a separate plugin / execution.
 This means that the signature tests will run during the maven `test` phase.
 
-You need to configure your test client with the system property `jimage.dir=<path-your-server-has-access-to>`.
+Your test client is configured with the property `jimage.dir=${project.build.directory}/jimage/`.
 This is because, when running the signature tests on JDK 9+ we need to convert the 
 JDK modules back into class files for signature testing.
 

--- a/tck-dist/src/main/assembly/assembly.xml
+++ b/tck-dist/src/main/assembly/assembly.xml
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!--
- ~ Copyright (c) 2022, 2023 Contributors to the Eclipse Foundation
+ ~ Copyright (c) 2022, 2024 Contributors to the Eclipse Foundation
  ~
  ~ This program and the accompanying materials are made available under the
  ~ terms of the Eclipse Public License v. 2.0, which is available at
@@ -53,6 +53,11 @@
         <include>**/*.txt</include>
       </includes>
     </fileSet>
+    <!-- The TCK source jar -->
+    <fileSet>
+      <directory>target/copied-sources/</directory>
+      <outputDirectory>artifacts/</outputDirectory>
+    </fileSet>
     <!-- The docs -->
     <fileSet>
       <directory>target/generated-docs</directory>
@@ -81,7 +86,6 @@
     <dependencySet>
       <includes>
         <include>jakarta.data:jakarta.data-tck</include>
-        <include>jakarta.data:jakarta.data-tck:jar:sources</include>
       </includes>
       <useTransitiveDependencies>true</useTransitiveDependencies>
       <outputDirectory>artifacts</outputDirectory>

--- a/tck-dist/src/main/pom.xml
+++ b/tck-dist/src/main/pom.xml
@@ -1,0 +1,37 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+ ~ Copyright (c) 2024 Contributors to the Eclipse Foundation
+ ~
+ ~ This program and the accompanying materials are made available under the
+ ~ terms of the Eclipse Public License v. 2.0, which is available at
+ ~ http://www.eclipse.org/legal/epl-2.0.
+ ~
+ ~ This Source Code may also be made available under the following Secondary
+ ~ Licenses when the conditions for such availability set forth in the
+ ~ Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+ ~ version 2 with the GNU Classpath Exception, which is available at
+ ~ https://www.gnu.org/software/classpath/license.html.
+ ~
+ ~ SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+  -->
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+ xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+ xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+
+ <modelVersion>4.0.0</modelVersion>
+ 
+ <!-- 
+      This is just an aggregator pom so that dependabot can find the starter poms
+      This aggregator is not packaged into the tck distribution
+  -->
+  
+ <groupId>jakarta.data</groupId>
+ <artifactId>data-tck-aggregator</artifactId>
+ <version>1.0-SNAPSHOT</version>
+  
+  <modules>
+	  <module>starter/ee-pom.xml</module>
+	  <module>starter/se-pom.xml</module>
+  </modules>
+ 
+ </project>

--- a/tck-dist/src/main/starter/ee-pom.xml
+++ b/tck-dist/src/main/starter/ee-pom.xml
@@ -37,13 +37,13 @@
   <jakarta.servlet.version>6.1.0</jakarta.servlet.version>
   <jakarta.enterprise.cdi.version>4.1.0</jakarta.enterprise.cdi.version>
 
-  <arquillian.version>1.7.0.Alpha13</arquillian.version>
-  <junit.version>5.9.0</junit.version>
+  <arquillian.version>1.8.0.Final</arquillian.version>
+  <junit.version>5.10.2</junit.version>
   <sigtest.version>2.3</sigtest.version>
 
   <maven.dep.plugin.version>3.3.0</maven.dep.plugin.version>
-  <maven.comp.plugin.version>3.10.1</maven.comp.plugin.version>
-  <maven.surefire.plugin.version>2.22.2</maven.surefire.plugin.version>
+  <maven.comp.plugin.version>3.13.0</maven.comp.plugin.version>
+  <maven.surefire.plugin.version>3.2.5</maven.surefire.plugin.version>
 
   <!-- Location to copy runtime dependencies to application server -->
   <application.server.lib>[path/to/appserver/lib]</application.server.lib>
@@ -121,12 +121,12 @@
   <dependency>
    <groupId>jakarta.servlet</groupId>
    <artifactId>jakarta.servlet-api</artifactId>
-   <version>6.0</version>
+   <version>${jakarta.servlet.version}</version>
   </dependency>
   <dependency>
    <groupId>jakarta.enterprise</groupId>
    <artifactId>jakarta.enterprise.cdi-api</artifactId>
-   <version>4.0</version>
+   <version>${jakarta.enterprise.cdi.version}</version>
   </dependency>
  </dependencies>
  <!-- end::testClientDep[] -->

--- a/tck-dist/src/main/starter/ee-pom.xml
+++ b/tck-dist/src/main/starter/ee-pom.xml
@@ -20,7 +20,7 @@
 
  <modelVersion>4.0.0</modelVersion>
 
- <groupId>[YOUR_GROUP_ID]</groupId>
+ <groupId>jakarta.data</groupId>
  <artifactId>data-tck-runner</artifactId>
  <version>1.0-SNAPSHOT</version>
  <name>Jakarta Data TCK Runner for [YOUR_IMPL]</name>
@@ -33,9 +33,9 @@
   <maven.compiler.testRelease>17</maven.compiler.testRelease>
 
   <!-- Dependency and Plugin Versions -->
-  
-  <!-- Default - use local snapshot - can be overwritten by runner -->
-  <jakarta.data.version>1.0.0-SNAPSHOT</jakarta.data.version>
+  <jakarta.data.version>1.0.0</jakarta.data.version>
+  <jakarta.servlet.version>6.1.0</jakarta.servlet.version>
+  <jakarta.enterprise.cdi.version>4.1.0</jakarta.enterprise.cdi.version>
 
   <arquillian.version>1.7.0.Alpha13</arquillian.version>
   <junit.version>5.9.0</junit.version>
@@ -52,8 +52,11 @@
    configuration -->
   <logging.config>${project.basedir}/logging.properties</logging.config>
 
-  <!-- Output directory -->
-  <targetDirectory>${project.basedir}/target</targetDirectory>
+  <!-- jimage output directory -->
+  <jimage.dir>${project.build.directory}/jimage/</jimage.dir>
+  
+  <!-- Default execution groups - user should override -->
+  <includedTests>core</includedTests>
  </properties>
 
  <repositories>
@@ -118,12 +121,12 @@
   <dependency>
    <groupId>jakarta.servlet</groupId>
    <artifactId>jakarta.servlet-api</artifactId>
-   <version>6.0.0</version>
+   <version>6.0</version>
   </dependency>
   <dependency>
    <groupId>jakarta.enterprise</groupId>
    <artifactId>jakarta.enterprise.cdi-api</artifactId>
-   <version>4.0.1</version>
+   <version>4.0</version>
   </dependency>
  </dependencies>
  <!-- end::testClientDep[] -->
@@ -164,11 +167,11 @@
       <tck_port>[TODO]</tck_port>
       <!-- end::arquillian[] -->
       <!-- tag::signature[] -->
-      <jimage.dir>[path-to-extract-jvm-classes]</jimage.dir>
-      <signature.sigTestClasspath>[path-to]/jakarta.data-api.jar:[path-to-jimage]/java.base:[path-to-jimage]/java.rmi:[path-to-jimage]/java.sql:[path-to-jimage]/java.naming</signature.sigTestClasspath>
+      <jimage.dir>${jimage.dir}</jimage.dir>
+      <signature.sigTestClasspath>[path-to]/jakarta.data-api.jar:${jimage.dir}/java.base:${jimage.dir}/java.rmi:${jimage.dir}/java.sql:${jimage.dir}/java.naming</signature.sigTestClasspath>
       <!-- end::signature[] -->
       <!-- tag::logging[] -->
-      <java.util.logging.config.file>${basedir}/../logging.properties</java.util.logging.config.file>
+      <java.util.logging.config.file>${logging.config}</java.util.logging.config.file>
       <!-- end::logging[] -->
       <!-- tag::profile[] -->
       <!-- Optional, Arquillian will deploy artifacts by default -->
@@ -180,10 +183,7 @@
          Profiles:[core|web|full] 
          Entities:[nosql|persistence] 
          Other:   [signature] -->
-     <groups>[TODO]</groups>
-     <!-- If running back-to-back tests at different levels 
-     use this to distinguish the results -->
-     <reportNameSuffix>[OPTIONAL]</reportNameSuffix>
+     <groups>${includedTests}</groups>
      <!-- Ensure surfire plugin looks under src/main/java 
      instead of src/test/java -->
      <testSourceDirectory>${basedir}/src/main/java/</testSourceDirectory>

--- a/tck-dist/src/main/starter/se-pom.xml
+++ b/tck-dist/src/main/starter/se-pom.xml
@@ -36,14 +36,14 @@
    <jakarta.servlet.version>6.1.0</jakarta.servlet.version>
   <jakarta.enterprise.cdi.version>4.1.0</jakarta.enterprise.cdi.version>
 
-  <arquillian.version>1.7.0.Alpha13</arquillian.version>
-  <junit.version>5.9.0</junit.version>
+  <arquillian.version>1.8.0.Final</arquillian.version>
+  <junit.version>5.10.2</junit.version>
   <shrinkwrap.version>1.2.6</shrinkwrap.version>
   <sigtest.version>2.3</sigtest.version>
   
   <maven.dep.plugin.version>3.3.0</maven.dep.plugin.version>
-  <maven.comp.plugin.version>3.10.1</maven.comp.plugin.version>
-  <maven.surefire.plugin.version>2.22.2</maven.surefire.plugin.version>
+  <maven.comp.plugin.version>3.13.0</maven.comp.plugin.version>
+  <maven.surefire.plugin.version>3.2.5</maven.surefire.plugin.version>
 
   <!-- Pointer to logging.properties file that has the java.util.logging 
    configuration -->
@@ -124,7 +124,7 @@
    <artifactId>jakarta.enterprise.cdi-api</artifactId>
    <version>${jakarta.enterprise.cdi.version}</version>
   </dependency>
-  <!-- TODO add a CDI implementation that is supported by your Jakarta Data implementation -->
+  <!-- TODO add a DI framework implementation that is supported by your Jakarta Data implementation -->
  </dependencies>
  <!-- end::runtimeDep[] -->
 

--- a/tck-dist/src/main/starter/se-pom.xml
+++ b/tck-dist/src/main/starter/se-pom.xml
@@ -20,7 +20,7 @@
 
  <modelVersion>4.0.0</modelVersion>
 
- <groupId>[YOUR_GROUP_ID]</groupId>
+ <groupId>jakarta.data</groupId>
  <artifactId>data-tck-runner</artifactId>
  <version>1.0-SNAPSHOT</version>
  <name>Jakarta Data TCK Runner for [YOUR_IMPL]</name>
@@ -28,19 +28,19 @@
  <properties>
   <!-- General properties -->
   <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-
   <maven.compiler.release>17</maven.compiler.release>
   <maven.compiler.testRelease>17</maven.compiler.testRelease>
 
   <!-- Dependency and Plugin Versions -->
-  
-  <!-- Default - use local snapshot - can be overwritten by runner -->
-  <jakarta.data.version>1.0.0-SNAPSHOT</jakarta.data.version>
+  <jakarta.data.version>1.0.0</jakarta.data.version>
+   <jakarta.servlet.version>6.1.0</jakarta.servlet.version>
+  <jakarta.enterprise.cdi.version>4.1.0</jakarta.enterprise.cdi.version>
 
   <arquillian.version>1.7.0.Alpha13</arquillian.version>
   <junit.version>5.9.0</junit.version>
+  <shrinkwrap.version>1.2.6</shrinkwrap.version>
   <sigtest.version>2.3</sigtest.version>
-
+  
   <maven.dep.plugin.version>3.3.0</maven.dep.plugin.version>
   <maven.comp.plugin.version>3.10.1</maven.comp.plugin.version>
   <maven.surefire.plugin.version>2.22.2</maven.surefire.plugin.version>
@@ -49,8 +49,11 @@
    configuration -->
   <logging.config>${project.basedir}/logging.properties</logging.config>
 
-  <!-- Output directory -->
-  <targetDirectory>${project.basedir}/target</targetDirectory>
+  <!-- jimage output directory -->
+  <jimage.dir>${project.build.directory}/jimage/</jimage.dir>
+  
+  <!-- Default execution groups - user should override -->
+  <includedTests>standalone</includedTests>
  </properties>
 
  <repositories>
@@ -99,28 +102,29 @@
    <artifactId>sigtest-maven-plugin</artifactId>
    <version>${sigtest.version}</version>
   </dependency>
-  <!-- APIs referenced by TCK that do not require implementations for standalone 
-   tests -->
+  <!-- APIs referenced by TCK that do not require implementations for standalone tests -->
   <dependency>
    <groupId>org.jboss.shrinkwrap</groupId>
    <artifactId>shrinkwrap-api</artifactId>
-   <version>1.2.6</version>
+   <version>${shrinkwrap.version}</version>
   </dependency>
   <dependency>
    <groupId>org.jboss.arquillian.junit5</groupId>
    <artifactId>arquillian-junit5-core</artifactId>
-   <version>1.7.0.Alpha13</version>
+   <version>${arquillian.version}</version>
   </dependency>
   <dependency>
    <groupId>jakarta.servlet</groupId>
    <artifactId>jakarta.servlet-api</artifactId>
-   <version>6.0.0</version>
+   <version>${jakarta.servlet.version}</version>
   </dependency>
+ <!-- APIs referenced by TCK that do require implementations for standalone tests -->
   <dependency>
    <groupId>jakarta.enterprise</groupId>
    <artifactId>jakarta.enterprise.cdi-api</artifactId>
-   <version>4.0.1</version>
+   <version>${jakarta.enterprise.cdi.version}</version>
   </dependency>
+  <!-- TODO add a CDI implementation that is supported by your Jakarta Data implementation -->
  </dependencies>
  <!-- end::runtimeDep[] -->
 
@@ -151,8 +155,8 @@
      <!-- tag::systemProperties[] -->
      <systemPropertyVariables>
       <!-- tag::signature[] -->
-      <jimage.dir>[path-to-extract-jvm-classes]</jimage.dir>
-      <signature.sigTestClasspath>[path-to]/jakarta.data-api.jar:[path-to-jimage]/java.base:[path-to-jimage]/java.rmi:[path-to-jimage]/java.sql:[path-to-jimage]/java.naming</signature.sigTestClasspath>
+      <jimage.dir>${jimage.dir}</jimage.dir>
+      <signature.sigTestClasspath>[path-to]/jakarta.data-api.jar:${jimage.dir}/java.base:${jimage.dir}/java.rmi:${jimage.dir}/java.sql:${jimage.dir}/java.naming</signature.sigTestClasspath>
       <!-- end::signature[] -->
       <!-- tag::logging[] -->
       <java.util.logging.config.file>${logging.config}</java.util.logging.config.file>
@@ -164,12 +168,10 @@
      </systemPropertyVariables>
      <!-- end::systemProperties[] -->
      <!-- Supported tags 
+         Profiles: [standalone]
          Entities:[nosql|persistence] 
          Other:   [signature] -->
-     <groups>standalone</groups>
-     <!-- If running back-to-back tests at different levels
-      use this to distinguish the results -->
-     <reportNameSuffix>[OPTIONAL]</reportNameSuffix>
+     <groups>${includedTests}</groups>
      <!-- Ensure surfire plugin looks under src/main/java
       instead of src/test/java -->
      <testSourceDirectory>${basedir}/src/main/java/</testSourceDirectory>

